### ISSUE TITLE
[Feature] Remove System.out or System.err and use Logger for everything

### ISSUE THREAD

**Issue by lucianoRM** [testng-team/testng#2892]
Hello! 

We are using TestNG as a framework for executing E2E tests as a service. Everything works great but I have noticed that the code has multiple places where instead of a Logger, System.out and System.err are used to print information to the screen. 

For example: `org.testng.internal.Version#displayBanner`

It would be really cool if every console output came from a Logger like you are using in other places. This way, we can add extra information to the log line via configuration (requestId for example).
Otherwise, it's really hard for us to find important information in our logs. For example, some failure stack traces are printed this way and are imposible to find.

**Comment by krmahadevan** [comment]
@lucianoRM what version are you working with? I believe TestNG 7.7.1 uses an slf4j api for its logging needs.

**Comment by krmahadevan** [comment]
@lucianoRM would you be willing to help raise a PR to fix it? We will be more than glad to help you get it merged

**Comment by lucianoRM** [comment]
Yeah! I could try to. I don't know if I have the time right now though. If I find a slot, I'll send the PR

**Comment by juherr** [comment]
Triage: **still relevant**.

Current master still has production `System.out`/`System.err` writes in multiple places, including `Version.displayBanner()`, `TestNG`, `TestNGContentHandler`, `SuiteRunnerWorker` and several built-in reporters. I found no merged/open PR that completes the requested migration.

The scope needs a maintainer decision before implementation: some writes are deliberately console/CLI output (for example the usage banner and text reporters), while others are diagnostics that could plausibly go through logging. Replacing every standard-stream write mechanically would therefore risk changing CLI/reporter contracts rather than just improving log routing.
